### PR TITLE
Jac/versioning and strings and some fixes

### PR DESCRIPTION
**Pull request by jacalata**
This looks enormous but it's ok!
- 50% of files are either string formatting in the localization files, or added tests
- 80% of the product code changes are just going from a hardcoded string to importing the translated strings. 

Product code changes
- some rearranged error handling, when the given error messages didn't match what we were saying 
- version number generated based on most recent tag + workspace condition (https://pypi.org/project/setuptools-scm/)
c:\dev\tabcmd2>tabcmd -v
tabcmd.exe - Tableau Server Command Line Utility v0.1.dev235+g44b1829.d20220607

### CODE REVIEW

File: .github/workflows/generate-metadata.yml [hunk 1]
```diff
@@ -8,50 +8,34 @@ on:
...
 
     steps:
     - uses: actions/checkout@v2
 
-    - name: Set up Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    - name: Set up Python
       uses: actions/setup-python@v2
       with:
-        python-version: ${{ matrix.python-version }}
+        python-version: '3.7'
```
Comment (chad-tableau) [inline .github/workflows/generate-metadata.yml:19]: Is this intentional?

**Review by chad-tableau — APPROVED** [review]
LGTM, but I'm not a python, or yaml, expert.

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+# https://data-dive.com/multi-os-deployment-in-cloud-using-pyinstaller-and-github-actions
+# https://anshumanfauzdar.medium.com/using-github-actions-to-bundle-python-application-into-a-single-package-and-automatic-release-834bd42e0670
+
+
+# TODO: actually plan when we want to trigger this job
+#  - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+on:
+  push:
+    tags: release
+    branches: jac/pyinstaller
```
Comment (mmuttreja-tableau) [inline .github/workflows/package.yml:17]: Does this break if `jac/pyinstaller` branch gets deleted?

Comment (jacalata) [reply .github/workflows/package.yml:17]: No, it's still triggered by any push tagged "release"

File: .github/workflows/generate-metadata.yml (hunk 1 above)
Comment (jacalata) [reply .github/workflows/generate-metadata.yml:19]: yes, this one doesn't need to run on more than one version. Should probably just make it 3

**Pull request merged**